esphome/components/climate_mitsubishi/climate_mitsubishi.cpp

```diff
@@ -0,0 +1,795 @@
+#include <chrono>
+#include <string>
+#include "esphome/components/climate/climate.h"
+#include "esphome/components/climate/climate_mode.h"
+#include "esphome/components/uart/uart.h"
+#include "esphome/components/sensor/sensor.h"
+#include "esphome/components/binary_sensor/binary_sensor.h"
+#include "esphome/components/select/select.h"
+
+#include "climate_mitsubishi.h"
+#include "mitsubishi_protocol.h"
+
+using namespace esphome::climate;
+
+namespace esphome {
+namespace climate_mitsubishi {
+
+static const char *const TAG = "mitsubishi.climate";
+constexpr size_t CONNECT_RETRY_INTERVAL_MS = 10000;
+constexpr size_t INFO_REQUEST_INTERVAL_MS = 400;
+constexpr size_t SETTINGS_REQUEST_INTERVAL_MS = 2000;
+constexpr size_t REQUEST_TIMEOUT_MS = 1000;
+
+ClimateMitsubishi::ClimateMitsubishi()
+    : compressor_frequency_sensor_(nullptr),
+      fan_velocity_sensor_(nullptr),
+      conflicted_sensor_(nullptr),
+      preheat_sensor_(nullptr),
+      control_temperature_sensor_(nullptr),
+      inject_enable_switch_(nullptr),
+      remote_temperature_number_(nullptr),
+      vertical_airflow_select_(nullptr),
+      high_precision_temp_setting_(true),
+      status_rotation_(0),
+      temperature_offset_(0) {
+  this->traits_ = climate::ClimateTraits();
+  this->traits_.set_supported_modes({climate::CLIMATE_MODE_OFF, climate::CLIMATE_MODE_COOL, climate::CLIMATE_MODE_HEAT,
+                                     climate::CLIMATE_MODE_FAN_ONLY, climate::CLIMATE_MODE_DRY,
+                                     climate::CLIMATE_MODE_HEAT_COOL});
+  this->traits_.set_supported_fan_modes({climate::CLIMATE_FAN_AUTO, climate::CLIMATE_FAN_QUIET});
+  this->traits_.set_supported_custom_fan_modes({"1", "2", "3", "4"});
+  this->traits_.set_supports_current_temperature(true);
+  this->traits_.set_supports_action(true);
+  this->traits_.set_visual_current_temperature_step(0.1);
+  this->traits_.set_visual_target_temperature_step(0.5);
+  this->pending_requests_ = std::list<RequestSlot *>();
+}
+
+ClimateTraits ClimateMitsubishi::traits() { return traits_; }
+
+ClimateMode ClimateMitsubishi::mode_to_climate_mode_(uint8_t mode) {
+  switch (mode) {
```

Comment:
```suggestion
  switch (mode > 0x08 ? mode - 0x08 : mode) {
```
For devices with iSee sensor and when it is enabled, then mode is shifted by 0x08.